service/block/event_test.go

```diff
@@ -31,9 +32,9 @@ func TestEventLoop(t *testing.T) {
 	numBlocks := 3
 	expectedBlocks := mockFetcher.generateBlocks(t, numBlocks)
 
-	ctx, cancel = context.WithCancel(context.Background())
-	t.Cleanup(cancel)
-
+	// as we calculate extended square twice, this may take a lot of time, causing the main thread expect the result
+	// earlier than its ready.
+	time.Sleep(100 * time.Millisecond)
```

Comment:
This seems a bit...brittle? Instead of sleeping, shouldn't there be some sort of async `await` or callback?

Reply:
Well, we just shouldn't calculate extended headers twice ;) But that's a whole different story...
Adding callbacks and/or awaiting on chan to API just to make test happier is not a good idea, though I understand where you are coming from.

Reply:
Agree with John on this: I think this kind of timeouts are the reason we experience non-deterministic test-flakiness in tendermint. No need to change anything here though. And as you said the square will only be computed once in the future (if we have an issue for that I would recommend linking this comment or code there).

Reply:
@renaynay, do we have an issue for this? I feel like you created one at some point?

Reply:
i think this https://github.com/celestiaorg/celestia-node/issues/125